File: JAVASCRIPT/array-missing-value.js
const ARRAY_LIMIT = 100;
const SEED        = new Date();

function main() {
	const arrayOne = new Array(ARRAY_LIMIT);
	const arrayTwo = new Array(ARRAY_LIMIT);
	
	let randomInteger, answer, i;

	randomInteger = Math.floor(
		(SEED * 12345 + Math.random()) % ARRAY_LIMIT
		);

	populateArrays(arrayOne, arrayTwo);
	modifyArray(arrayTwo, randomInteger);

	answer = checkDifference(arrayOne, arrayTwo);

	console.log(`
\r\nFIND THE RANDOM MISSING VALUE\n
=============================\n
Array 1 | Array 2`);
	console.log(`--------+--------`);


	for (i = 0; i < ARRAY_LIMIT; ++i) {
		console.log(`${arrayOne[i]}\t|\t${arrayTwo[i]}`);
	}

	console.log(`--------+--------\n`);
	console.log(`The random missing value is ${randomInteger}`);
	console.log(`And the missing vlaue from the modified array is ${answer}`);
}

function populateArrays(arrayInitial = [], arrayModified = []) {
	let i;

	for (i = 0; i < ARRAY_LIMIT; ++i) {
		arrayInitial[i] = i + 1;
		arrayModified[i] = i + 1;
	}
}

function modifyArray(arrayNew = [], random = 0) {
	arrayNew[random] = 0;
}

function sumArray(array = []) {
	let i, sum;
	sum = 0;

	for (i = 0; i < ARRAY_LIMIT; ++i)
		sum += array[i];

	return sum;
}

function checkDifference(arrayInitial = [], arrayModified = []) {
	let sumInitial, sumModified, difference;

	sumInitial  = sumArray(arrayInitial);
	sumModified = sumArray(arrayModified);
	difference  = sumInitial - sumModified;

	return difference;
}

module.exports = main;

File: PYTHON/array_missing_value.py
from random import *

RANDOM_INT = randint(0,99)

def main():
	array_one = [0]*100
	array_two = [0]*100

	array_one = populate_array(array_one)
	array_two = populate_array(array_two)

	array_two = modify_array(array_two)

	answer = check_difference(array_one, array_two)

	print "\r\nFIND THE RANDOM MISSING VALUE"
	print "============================="
	print " "
	print " Array One\t|\tArray Two"
	print "----------------|---------------"

	for i in range(99):
		print str(i) + " : " + str(array_one[i]) + "\t\t|\t" + str(i) + " : " + str(array_two[i])

	print "--------------------------------"

	print "\nThe random value is: ", RANDOM_INT
	print "And the missing value from the modified array is: ", answer, "\r\n"

def populate_array(array = []):
	for num in range(100):
		array[num] = num
	return array
	
def modify_array(array = []):
	array[RANDOM_INT] = 0

	return array

def sum_array(array = []):
	sum = 0

	for num in array:
		sum += num

	return sum

def check_difference(array_initial = [], array_modified = []):
	sum_initial = 0
	sum_modified = 0
	difference = 0

	sum_initial = sum_array(array_initial)
	sum_modified = sum_array(array_modified)
	difference = sum_initial - sum_modified

	return difference

if __name__ == "__main__":
	main()
